freqtrade/freqtradebot.py

```diff
@@ -946,7 +946,7 @@ def _check_and_execute_sell(self, trade: Trade, sell_rate: float,
 
         if should_sell.sell_flag:
             logger.info(f'Executing Sell for {trade.pair}. Reason: {should_sell.sell_type}')
-            self.execute_sell(trade, sell_rate, should_sell.sell_type)
+            self.execute_sell(trade, sell_rate, should_sell.sell_type, should_sell.sell_reason)
```

Comment:
Maybe we can have the tuple (`should_sell / SellCheckTuple`) as parameter here instead of using 2 of the 3 provided values seperately?

Reply:
Will do 👍🏻 But now that im looking at the code i see `SellCheckTuple` having bool + `SellType`. However each time `sell_flag=False` we also have `sell_type=SellType.NONE` and each time `sell_flag=True` then `sell_type` is something other than `SellType.NONE`. It seems like `sell_flag` is redundant here. I think we should get rid of it and simply have `SellCheckTuple` contain `SellType` + `str` reason. Any objections?

Reply:
yes, that'll be fine and should be a nice improvement too 👍